homeassistant/components/matter/update.py

```diff
@@ -0,0 +1,229 @@
+"""Matter update."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Any
+
+from chip.clusters import Objects as clusters
+from matter_server.common.errors import UpdateCheckError, UpdateError
+from matter_server.common.models import MatterSoftwareVersion
+
+from homeassistant.components.update import (
+    ATTR_LATEST_VERSION,
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityDescription,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.restore_state import ExtraStoredData
+
+from .entity import MatterEntity
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+SCAN_INTERVAL = timedelta(hours=12)
+POLL_AFTER_INSTALL = 10
+
+ATTR_SOFTWARE_UPDATE = "software_update"
+
+
+@dataclass
+class MatterUpdateExtraStoredData(ExtraStoredData):
+    """Extra stored data for Matter node firmware update entity."""
+
+    software_update: MatterSoftwareVersion | None = None
+
+    def as_dict(self) -> dict[str, Any]:
+        """Return a dict representation of the extra data."""
+        return {
+            ATTR_SOFTWARE_UPDATE: self.software_update.as_dict()
+            if self.software_update is not None
+            else None,
+        }
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> MatterUpdateExtraStoredData:
+        """Initialize the extra data from a dict."""
+        if data[ATTR_SOFTWARE_UPDATE] is None:
+            return cls()
+        return cls(MatterSoftwareVersion.from_dict(data[ATTR_SOFTWARE_UPDATE]))
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter lock from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.UPDATE, async_add_entities)
+
+
+class MatterUpdate(MatterEntity, UpdateEntity):
+    """Representation of a Matter node capable of updating."""
+
+    _attr_should_poll = True
```

Comment:
Maybe explain why the update entity is polled

Reply:
Added. Thanks for the review!